frontend/src/views/WheresMyGene/components/GeneSearchBar/index.tsx

```diff
@@ -160,35 +211,124 @@ export default function GeneSearchBar({ onGenesChange }: Props): JSX.Element {
+      return;
+    }
+    setOpen(false);
+  };
+  const handleChange = (_, newValue: DefaultMenuSelectOption[] | null) => {
+    return setSelectedGenes(newValue as Gene[]);
+  };
+  const handleClick = () => {
+    setOpen(true);
+  };
+
+  const handlePaste = () => {
+    setPendingPaste(true);
+  };
+
+  const handleEnter = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === "Enter" && pendingPaste) {
+      event.preventDefault();
+      const newSelectedGenes = [...selectedGenes];
+      const pastedGenes = pull(uniq(input.split(/[ ,]+/)), "");
+      const allGenes = genes.reduce((acc, gene) => {
+        return acc.set(gene.name, gene);
+      }, new Map<Gene["name"], Gene>());
+      pastedGenes.map((gene) => {
+        const newGene = allGenes.get(gene);
+        if (!newGene) {
+          Toast.show({
+            intent: Intent.DANGER,
+            message: `Gene not found: ${gene}`,
+          });
+        } else if (!newSelectedGenes.includes(newGene))
+          newSelectedGenes.push(newGene);
+      });
+      setPendingPaste(false);
+      setOpen(false);
+      return setSelectedGenes(newSelectedGenes);
+    }
+  };
+
+  const handleInput = (
+    _,
+    value: string,
+    reason: AutocompleteInputChangeReason
+  ) => {
+    if (reason === "input") setInput(value);
+  };
+
+  const useStyles = makeStyles((theme: Theme) => {
+    const colors = getColors({ theme });
+    const shadows = getShadows({ theme });
+    const corners = getCorners({ theme });
+    return {
+      paper: {
+        boxShadow: "none",
+        margin: 0,
+      },
+      popper: {
+        backgroundColor: "white",
+        border: `1px solid ${colors?.gray[100]}`,
+        borderRadius: corners?.m,
+        boxShadow: shadows?.m,
+        color: "#586069",
+        fontSize: 13,
+        width: 377,
+        zIndex: 3, // The x axis wrapper is set at 2
+      },
+      popperDisablePortal: {
+        position: "relative",
+        width: "100% !important",
+      },
+    };
+  });
+
+  const classes = useStyles();
+
+  const ref = useRef(null);
 
   return (
     <Container>
       <GeneSets onSelect={handleGenesetsSelect} />
 
       <br />
       <br />
-
-      <label htmlFor="first-n-genes">Select first N genes</label>
-      <br />
-      <input id="first-n-genes" onChange={handleInputChange} value={input} />
-      <MultiSelectWrapper>
-        <MultiSelect
-          itemPredicate={itemPredicate}
-          onItemSelect={handleItemSelect}
-          onRemove={handleItemRemove}
-          items={genes}
-          itemRenderer={renderItem}
-          tagRenderer={TagRenderer}
-          itemsEqual={areGenesEqual}
-          selectedItems={selectedGenes}
-          itemListRenderer={itemListRenderer}
+      <ButtonBase disableRipple onClick={handleClick} ref={ref}>
+        <span>Add Genes</span>
+      </ButtonBase>
+
+      <Popper open={open} className={classes.popper} anchorEl={ref.current}>
+        <MenuSelect
```

Comment:
Do you think we can generalize this component for tissue selector too?

Reply:
Good idea! I can pull it out now, or address in a separate PR. Thoughts?

Reply:
Ooh yeah that sounds good! Do you wanna take https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/single-cell-data-portal/1934 and just extract and use it then? Thank you!

Reply:
Sounds good!